RobotLocomotion/drake
# Modified Drake Build System to Support ROS If ROS Is Present.

Contributes to: https://github.com/RobotLocomotion/drake/issues/1733

_Again, this is in the spirit of cultivating a culture of small pull requests!_

# Major Changes
* Added a unit test called `ros_test` that demonstrates that it can instantiate a ROS node.

# Notes
* The unit test is automatically built if ROS is found on the local machine.
* The unit test will be extended to evaluate more drake-specific functionality in future PRs.

# How to Run Unit Test
_The following instructions only work on Ubuntu Linux 14.04._

[Install ROS Indigo](http://wiki.ros.org/indigo/Installation/Ubuntu).

Open two terminals. In the first terminal, start `roscore`:

```
$ source /opt/ros/indigo/setup.bash
$ roscore
```

In the second terminal, run the unit test:

```
$ source /opt/ros/indigo/setup.bash
$ cd [drake distro]/drake/pod-build
$ ctest -VV -R ros_test
```

Note that the unit test will fail if `roscore` is not running. This is expected behavior.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/2103)
